tests/test_uploads.py

```diff
@@ -1,209 +1,208 @@
+                'upload_paths': [full_path('single_file'),
+                                 full_path('one_symlink')]}
+
+        self.assertEqual(Submit(args).get_upload_files(),[
+            full_path('single_file/foo'),
+            full_path('one_symlink/nnn'),
+            full_path('one_symlink/many_subdirs/bar'),
+            full_path('one_symlink/many_subdirs/one'),
+            full_path('one_symlink/many_subdirs/a_subdir/adsf'),
+            full_path('one_symlink/many_subdirs/a_subdir/three'),
+            full_path('one_symlink/many_subdirs/a_subdir/two'),
+            full_path('one_symlink/many_subdirs/a_subdir/another_subdir/oimsdf'),
+            full_path('one_symlink/many_subdirs/a_subdir/another_subdir/even_another_subdir/ppp')])
+
+
+        # test upload_file and upload_paths together
+        upload_file = open(upload_file_path, 'w')
+        upload_file.write(full_path('single_file'))
+        upload_file.close()
+        args = {'cmd': 'some command',
+                'upload_only': True,
+                'upload_file': upload_file_path,
+                'upload_paths': [full_path('one_symlink')]}
+
+        self.assertEqual(Submit(args).get_upload_files(),[
+            full_path('single_file/foo'),
+            full_path('one_symlink/nnn'),
+            full_path('one_symlink/many_subdirs/bar'),
+            full_path('one_symlink/many_subdirs/one'),
+            full_path('one_symlink/many_subdirs/a_subdir/adsf'),
+            full_path('one_symlink/many_subdirs/a_subdir/three'),
+            full_path('one_symlink/many_subdirs/a_subdir/two'),
+            full_path('one_symlink/many_subdirs/a_subdir/another_subdir/oimsdf'),
+            full_path('one_symlink/many_subdirs/a_subdir/another_subdir/even_another_subdir/ppp')])
+
+
+    def test_make_request(self):
+        """This test depends on an object in :
+          gs://conductor-test/accounts/testing/files/myObject
+        with content:
+          'hi'
+        """
+        base_url = 'http://test.conductor.io:8080'
+
+        try:
+            uri_path = '/api/files/ping'
+            job = make_request(uri_path)
+        except Exception:
+            message = str(traceback.print_exc())
+            message += "could not connect to app. are you running a dev env?"
+            self.fail(message)
+
+
+        # this should produce a 400
+        uri_path = '/api/files/stat'
+        with self.assertRaises(HTTPError):
+        # with self.assertRaises(HttpError):
+            job = make_request(uri_path=uri_path)
+
+
+        uri_path = '/api/files/stat'
+        params = {'filename':'myObject'}
+        job = make_request(uri_path=uri_path,params=params)
+
+
+    def test_get_upload_url(self):
+        args = {'url': 'http://test.conductor.io:8080' }
+
+        uploader = Uploader(args)
+        self.assertEqual(uploader.__class__.__name__,'Uploader')
+
+        file_name = full_path('upload_file1')
+        md5= uploader.get_md5(file_name)

```

Comment:
I'm surprised `md5=` wasn't picked up by Codacy!

Reply:
It might be because the md5 module isn't imported in this file - although I agree that ideally it would catch any variable name stopping that includes modules / keywords that are part of the Python Standard Lib.

Reply:
I was referring to the missing white space between `md5` and `=`; IIRC, Codacy reports missing white spaces between math operators.